### PR TITLE
feat(cli): add support for DDB lambda trigger mocking

### PR DESCRIPTION
**Pull request by phani-srikar**
_Issue #, if available:_

_Description of changes:_
Add support for mocking the lambda triggers attached to `@model` types defined in customer's schema. 
These triggers are added following the CLI flow as noted [here](https://docs.amplify.aws/cli/usage/lambda-triggers/#as-a-part-of-the-graphql-api-types-with-model-annotation).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by offlineprogrammer — APPROVED** [review]

File: src/pages/cli/usage/mock.mdx [hunk 1]
```diff
@@ -46,6 +46,17 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
 - [@hasOne, @hasMany, @belongsTo, @manyToMany](/cli/graphql/data-modeling)
 - [@function](/cli/graphql/custom-business-logic#lambda-function-resolver)
 
+If you have any DynamoDB lambda triggers already setup on any of the `@model` types in your GraphQL schema, by following steps [listed here](/cli/usage/lambda-triggers/#as-a-part-of-the-graphql-api-types-with-model-annotation),
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:49]:
```suggestion
If you have DynamoDB Lambda triggers setup on `@model` types in your GraphQL schema, by following steps [listed here](/cli/usage/lambda-triggers/#as-a-part-of-the-graphql-api-types-with-model-annotation),
```

File: src/pages/cli/usage/mock.mdx [hunk 2]
```diff
@@ -46,6 +46,17 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
...
+you can test those lambda triggers locally via `amplify mock` or `amplify mock api` commands.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:50]: Removed those this. As it is collapsed above.

File: src/pages/cli/usage/mock.mdx [hunk 3]
```diff
@@ -46,6 +46,17 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
 - [@hasOne, @hasMany, @belongsTo, @manyToMany](/cli/graphql/data-modeling)
 - [@function](/cli/graphql/custom-business-logic#lambda-function-resolver)
 
+If you have any DynamoDB lambda triggers already setup on any of the `@model` types in your GraphQL schema, by following steps [listed here](/cli/usage/lambda-triggers/#as-a-part-of-the-graphql-api-types-with-model-annotation),
+you can test those lambda triggers locally via `amplify mock` or `amplify mock api` commands. 
+
+When a `CRUD` operation performed on the `@model` type using either the local DynamoDB endpoint (`http://localhost:62224`), or the local AppSync console (should be `http://localhost:20002`), 
+you can see that the connected lambda trigger in your project will be automatically invoked locally with the event data. The structure of the event that is sent to the lambda trigger 
+can be found [here](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_GetRecords.html).
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:54]:
```suggestion
The connected Lambda triggers are automatically invoked locally if you perform a CRUD operation on the `@model` type using either the local DynamoDB endpoint (`http://localhost:62224`) or the local AppSync console (should be `http://localhost:20002`). The structure of the event that is sent to the lambda trigger can be found [here](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_GetRecords.html). 
```

File: src/pages/cli/usage/mock.mdx [hunk 4]
```diff
@@ -46,6 +46,17 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
...
+you can test those lambda triggers locally via `amplify mock` or `amplify mock api` commands. 
+
+When a `CRUD` operation performed on the `@model` type using either the local DynamoDB endpoint (`http://localhost:62224`), or the local AppSync console (should be `http://localhost:20002`), 
+you can see that the connected lambda trigger in your project will be automatically invoked locally with the event data. The structure of the event that is sent to the lambda trigger 
+can be found [here](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_GetRecords.html). 
+
+The environment variables that are listed below in [Function mock environment variables](/cli/usage/mock/#function-mock-environment-variables), will be available for each invocation of the lambda trigger.
+In addition, if you wish to override some of the environment variables for local mocking, you can do so using a .env file within the function directory (ie. `<project root>/amplify/backend/function/<function name>/.env`)
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:57]:
```suggestion
The environment variables listed in [Function mock environment variables](/cli/usage/mock/#function-mock-environment-variables) are available for each invocation of the Lambda trigger.
In addition, you can use a .env file within the function directory (ie. `<project root>/amplify/backend/function/<function name>/.env`) to override any environment variables for local mocking.
```

File: src/pages/cli/usage/mock.mdx [hunk 5]
```diff
@@ -46,6 +46,17 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
 - [@hasOne, @hasMany, @belongsTo, @manyToMany](/cli/graphql/data-modeling)
 - [@function](/cli/graphql/custom-business-logic#lambda-function-resolver)
 
+If you have DynamoDB Lambda triggers setup on `@model` types in your GraphQL schema, by following steps [listed here](/cli/usage/lambda-triggers/#as-a-part-of-the-graphql-api-types-with-model-annotation),
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:49]:
```suggestion
If you have DynamoDB Lambda triggers set up on `@model` types in your GraphQL schema, by following steps [listed here](/cli/usage/lambda-triggers/#as-a-part-of-the-graphql-api-types-with-model-annotation),
```

File: src/pages/cli/usage/mock.mdx [hunk 6]
```diff
@@ -46,6 +46,17 @@ When defining a schema you can use directives from the GraphQL Transformer in lo
...
+then you can test those Lambda triggers locally via `amplify mock` or `amplify mock api` commands.
```
Comment (renebrandel) [inline src/pages/cli/usage/mock.mdx:50]:
```suggestion
then you can test those Lambda triggers locally via `amplify mock` or `amplify mock api`. 
```

**Review by renebrandel — APPROVED** [review]

**Pull request merged**